cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java

```
@@ -56,129 +56,121 @@ public AvroFileReader(Schema schema) {
 
-          // Start reading events from file
-          int count = 0;
-          long prevTimestamp = -1;
-          while (dataFileReader.hasNext()) {
-            loggingEvent = LoggingEvent.decode(dataFileReader.next(datum));
-            if (loggingEvent.getTimeStamp() >= fromTimeMs && logFilter.match(loggingEvent)) {
-              ++count;
-              if ((count > maxEvents || loggingEvent.getTimeStamp() >= toTimeMs)
-                && loggingEvent.getTimeStamp() != prevTimestamp) {
-                break;
-              }
-              callback.handle(new LogEvent(loggingEvent,
-                                           new LogOffset(LogOffset.INVALID_KAFKA_OFFSET, loggingEvent.getTimeStamp())));
+        // Start reading events from file
+        int count = 0;
+        long prevTimestamp = -1;
+        while (dataFileReader.hasNext()) {
+          loggingEvent = LoggingEvent.decode(dataFileReader.next(datum));
+          if (loggingEvent.getTimeStamp() >= fromTimeMs && logFilter.match(loggingEvent)) {
+            ++count;
+            if ((count > maxEvents || loggingEvent.getTimeStamp() >= toTimeMs)
+              && loggingEvent.getTimeStamp() != prevTimestamp) {
+              break;
             }
-            prevTimestamp = loggingEvent.getTimeStamp();
+            callback.handle(new LogEvent(loggingEvent,
+                                         new LogOffset(LogOffset.INVALID_KAFKA_OFFSET, loggingEvent.getTimeStamp())));
           }
+          prevTimestamp = loggingEvent.getTimeStamp();
         }
-      } finally {
-        try {
-          dataFileReader.close();
-        } catch (IOException e) {
-          LOG.error("Got exception while closing log file {}", file, e);
-        }
       }
-    } catch (Exception e) {
-      LOG.error("Got exception while reading log file {}", file, e);
-      throw Throwables.propagate(e);
+    } finally {
+      try {
+        dataFileReader.close();
```

Does `DataFileReader` implements `Closeable`? If it does, we should use the `try (resource)` syntax instead of closing it in finally.

---

If try-with-resources syntax is used then how do we ignore exception from `close()` method?

---

why we want to ignore the exception on close?

---

Since an exception during close of a file does not affect the log events read from the file, we ignore exceptions during close.

---

As a part of https://issues.cask.co/browse/CDAP-5080, I'm changing AvroFileReader to not pass the log events via a Callback, but instead, it will return a CloseableIterator of log events.
Because of that, the file close will be asynchronous (and so it doesn't need to be try-catch) here, since that's going to change anyways.
Because of that, the file